packages/enzyme/src/ShallowWrapper.js

```diff
@@ -402,7 +391,7 @@ class ShallowWrapper {
         instance.setState(state, callback);
         if (
           shouldRender &&
-          this[OPTIONS].lifecycleExperimental &&
+          !this[OPTIONS].disableLifecycleMethods &&
```

Comment:
This change could be the cause for the undefined in the context, my comment above. But it makes sense at first sight.

Reply:
This will be fixed at #1261